examples/dataservice-read/example.cpp

```diff
@@ -34,12 +34,71 @@
 namespace {
 const std::string gKeyId("");      // your here.access.key.id
 const std::string gKeySecret("");  // your here.access.key.secret
-const std::string gCatalogHRN(
-    "hrn:here:data:::here-optimized-map-for-visualization-2");
-const std::string gLayerId("omv-base-v2");
-const std::string gPartitionId("100000093");  // Helsinki, Finland
+const std::string gCatalogHRN("hrn:here:data:::edge-example-catalog");
+const size_t gMaxLayers(5);
+const size_t gMaxPartitions(5);
+const char gLogLabel[] = "read-example";
 }  // namespace
 
+std::string handleCatalogResponse(
+    const olp::dataservice::read::CatalogResponse& catalogResponse) {
+  std::string firstPartitionId;
+  if (catalogResponse.IsSuccessful()) {
+    const olp::dataservice::read::CatalogResult& responseResult =
+        catalogResponse.GetResult();
+    LOG_INFO_F(gLogLabel, "Catalog description: %s",
+               responseResult.GetDescription().c_str());
+    auto layers = catalogResponse.GetResult().GetLayers();
+    if (!layers.empty()) {
+      firstPartitionId = layers.front().GetId();
+    }
+    auto end = layers.size() <= gMaxLayers ? layers.end()
+                                           : layers.begin() + gMaxLayers;
+    for (auto it = layers.cbegin(); it != end; ++it) {
+      LOG_INFO_F(gLogLabel, "Layer '%s' (%s): %s", it->GetId().c_str(),
+                 it->GetLayerType().c_str(), it->GetDescription().c_str());
+    }
+  } else {
+    LOG_ERROR(gLogLabel, "Request catalog metadata - Failure");
+  }
+  return firstPartitionId;
+}
+
+std::string handlePartitionsResponse(
+    const olp::dataservice::read::PartitionsResponse& partitionsResponse) {
+  std::string firstLayerId;
+  if (partitionsResponse.IsSuccessful()) {
+    const olp::dataservice::read::PartitionsResult& responseResult =
+        partitionsResponse.GetResult();
+    auto partitions = responseResult.GetPartitions();
+    LOG_INFO_F(gLogLabel, "Layer contains %ld partitions.", partitions.size());
+    if (!partitions.empty()) {
```

Comment:
please separate with a newline to make it a little bit more visible

Reply:
done